### PR TITLE
[TT-9532] Added opentelemetry support

### PR DESCRIPTION
**Pull request by komalsukhani**
## Description

Added opentelemetry support in gateway component chart and all umbrella charts

## Related Issue

[TT-9532](https://tyktech.atlassian.net/browse/TT-9532)

## Test Coverage For This Change

Tested with Jaeger

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

[TT-9532]: https://tyktech.atlassian.net/browse/TT-9532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-data-plane/values.yaml [hunk 1]
```diff
@@ -304,6 +304,53 @@ tyk-gateway:
...
+      # Flag that can be used to enable TLS
+        enabled: false
+      # Flag that can be used to skip TLS verification if TLS is enabled
+        insecureSkipVerify: true
+      # Maximum TLS version that is supported.
+        maxVersion: 1.3
+      # Minimum TLS version that is supported
+        minVersion: 1.2
+      # Path to the cert file
+        certFileName:
```
Comment (buraksekili) [inline tyk-data-plane/values.yaml:338]: Should we set it to an empty string, in order to indicate that the value is nil? What do you think?
```suggestion
        certFileName: ""
```

File: tyk-data-plane/values.yaml [hunk 2]
```diff
@@ -304,6 +304,53 @@ tyk-gateway:
...
+      # Path to the key file  
+        keyFileName:
```
Comment (buraksekili) [inline tyk-data-plane/values.yaml:340]: Should we set it to an empty string, in order to indicate that the value is nil? What do you think?
```suggestion
        keyFileName: ""
```

File: tyk-data-plane/values.yaml [hunk 3]
```diff
@@ -304,6 +304,53 @@ tyk-gateway:
...
+      # Path to CA file
+        caFileName:
```
Comment (buraksekili) [inline tyk-data-plane/values.yaml:342]: Should we set it to an empty string, in order to indicate that the value is nil? What do you think?
```suggestion
        caFileName: ""
```

File: tyk-data-plane/values.yaml [hunk 4]
```diff
@@ -304,6 +304,53 @@ tyk-gateway:
...
+      # Existing secret that stores TLS and CA Certificate
+        certificateSecretName:
```
Comment (buraksekili) [inline tyk-data-plane/values.yaml:344]: Should we set it to an empty string, in order to indicate that the value is nil? What do you think?
```suggestion
        certificateSecretName: ""
```

File: components/tyk-gateway/values.yaml [hunk 5]
```diff
@@ -309,3 +309,52 @@ gateway:
   # used to decide whether to send the results back directly to Tyk without a hybrid pump
   # if you want to send analytics to control plane instead of pump, change analyticsConfigType to "rpc"
   analyticsConfigType: ""
+
+  # opentelemetry is used to configure opentelemetry for Tyk Gateway
+  opentelemetry:
+  # Used to enable/disable opentelemetry
```
Comment (buraksekili) [inline components/tyk-gateway/values.yaml:315]: Can you please update indentation of the comments?

File: tyk-data-plane/README.md [hunk 6]
```diff
@@ -334,6 +334,13 @@ tyk-gateway:
               averageValue: 10000m
 ```
 
+#### Enable OpenTelemetry
+To enable opentelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
+
+You can also configure connection settings for it's exporter. By default `grpc` exporter is enabled on `localhost:4317` endpoint.
+
+ To enable TLS settings for exporter, you can set`gateway.opentelemetry.tls.enabled` to true.
```
Comment (buraksekili) [inline tyk-data-plane/README.md:342]:
```suggestion
 To enable TLS settings for the exporter, you can set `gateway.opentelemetry.tls.enabled` to true. 
```

File: tyk-oss/README.md [hunk 7]
```diff
@@ -333,6 +333,13 @@ tyk-gateway:
               averageValue: 10000m
 ```
 
+#### OpenTelemetry
+To enable opentelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
+
+You can also configure connection settings for it's exporter. By default `grpc` exporter is enabled on `localhost:4317` endpoint.
+
+ To enable TLS settings for exporter, you can set`gateway.opentelemetry.tls.enabled` to true.
```
Comment (buraksekili) [inline tyk-oss/README.md:341]:
```suggestion
 To enable TLS settings for the exporter, you can set `gateway.opentelemetry.tls.enabled` to true. 
```

**Review by buraksekili — COMMENTED** [review]
I just left minor comments. Overall look good to me. also, thank you for inline documentation and comments

File: tyk-data-plane/README.md [hunk 8]
```diff
@@ -334,6 +334,13 @@ tyk-gateway:
               averageValue: 10000m
 ```
 
+#### Enable OpenTelemetry
+To enable opentelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
```
Comment (buraksekili) [inline tyk-data-plane/README.md:338]:
```suggestion
To enable OpenTelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
```

File: tyk-stack/README.md [hunk 9]
```diff
@@ -311,6 +311,13 @@ tyk-gateway:
               averageValue: 10000m
 ```
 
+#### OpenTelemetry
+To enable opentelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
```
Comment (buraksekili) [inline tyk-stack/README.md:315]:
```suggestion
To enable OpenTelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
```

File: tyk-stack/README.md [hunk 10]
```diff
@@ -311,6 +311,13 @@ tyk-gateway:
...
+
+You can also configure connection settings for it's exporter. By default `grpc` exporter is enabled on `localhost:4317` endpoint.
+
+ To enable TLS settings for exporter, you can set`gateway.opentelemetry.tls.enabled` to true.
```
Comment (buraksekili) [inline tyk-stack/README.md:319]:
```suggestion
 To enable TLS settings for the exporter, you can set `gateway.opentelemetry.tls.enabled` to true. 
```

File: tyk-oss/README.md [hunk 11]
```diff
@@ -333,6 +333,13 @@ tyk-gateway:
               averageValue: 10000m
 ```
 
+#### OpenTelemetry
+To enable opentelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
```
Comment (buraksekili) [inline tyk-oss/README.md:337]:
```suggestion
To enable OpenTelemetry for Gateway set `gateway.opentelemetry.enabled` flag to true. It is disabled by default.
```

**Review by buraksekili — COMMENTED** [review]
Do we also need to update tyk-docs?

**Comment by komalsukhani** [comment]
@buraksekili Yes. It's in my TODO. Will do that probably new week.

**Review by buraksekili — APPROVED** [review]

**Pull request merged**